### PR TITLE
Acyclic monad draft implementation

### PR DESCRIPTION
**Pull request by adithyaov**
Although, I don't think there will be much use of Acyclic Monads with the current implementation. Maybe this can be extended in the near future.

### CODE REVIEW

File: AUTHORS.md [hunk 1]
```diff
@@ -4,6 +4,7 @@ The Alga library was originally developed by
 
 but over time many contributors helped make it much better, including (among others):
 
+* [Adithya Obilisetty](mailto:adi.obilisetty@gmail.com) [@adithyaov](https://github.com/adithyaov)
```
Comment (adithyaov) [inline AUTHORS.md]: This file should not have changed.

**Comment by snowleopard** [comment]
Can you give some examples of how this works?

**Comment by adithyaov** [comment]
@snowleopard 
```haskell
graph = runAcyclicMonad  $ do
  a <- 1 `vertexFrom` []
  b <- 2 `vertexFrom` [a]
  c <- 1 `vertexFrom` [a, b] -- Computation ignored.
  d <- 4 `vertexForm` [c]
  return ()
```
would result in a graph with edges `[(2, 1), (4, 1)]`
Please ignore this implementation for now. I think there could be a better implementation.

I'll experiment a bit and get back to you.

**Comment by adithyaov** [comment]
There are better ways to produce Acyclic Graphs. Closing this PR for now.

**Pull request closed without merging**